libs/providers/flagd-web/src/lib/local-storage-flagd-cache.ts

```diff
@@ -0,0 +1,98 @@
+import { EvaluationContext, FlagValue, ResolutionDetails } from '@openfeature/js-sdk';
+import { Md5 } from 'ts-md5';
+import { FlagdCache } from './flagd-cache';
+
+type CacheItem<T = FlagValue> = {
+  expiry: number;
+  details: ResolutionDetails<T>;
+  variant: string;
+};
+
+const PREXFIX = '@OpenFeature/flagd-';
+const SEPARATOR = '--';
+const DEFAULT_OPTIONS = {
+  expiryCheckInterval: 60000,
+  itemTtl: 300,
+};
+
+export interface Options {
+  expiryCheckInterval: number;
+  itemTtl: number;
+}
+
+function getOptions(options: Partial<Options> = {}) {
+  return {
+    ...DEFAULT_OPTIONS,
+    ...options,
+  };
+}
+
+export class LocalStorageFlagdCache implements FlagdCache {
+  private _ttl: number | undefined;
+
+  constructor(options?: Partial<Options>) {
+    const defaultedOptions = getOptions(options);
+    this._ttl = defaultedOptions.itemTtl === 0 ? undefined : defaultedOptions.itemTtl;
+    setInterval(() => {
+      const currentTimestamp = new Date().getTime();
+
+      Object.entries(localStorage).forEach((each) => {
+        if (each[0].startsWith(PREXFIX)) {
+          const item: CacheItem = JSON.parse(each[1]);
+          if (currentTimestamp > item.expiry) {
+            localStorage.removeItem(each[0]);
+          }
+        }
+      });
+    }, defaultedOptions.expiryCheckInterval);
+  }
+
+  private buildKey(flagKey: string, context: EvaluationContext) {
+    return `${PREXFIX}${flagKey}${SEPARATOR}${Md5.hashStr(JSON.stringify(context))}`;
+  }
+
+  set<T extends FlagValue>(key: string, context: EvaluationContext, details: ResolutionDetails<T>): Promise<void> {
+    const cacheKey = this.buildKey(key, context);
+    localStorage.setItem(
+      cacheKey,
+      JSON.stringify({
+        expiry: this._ttl ? new Date(new Date().getTime() + this._ttl).getTime() : undefined,
+        details: { ...details, reason: 'CACHED' }
+      })
+    );
+    return Promise.resolve();
+  }
+
+  get<T extends FlagValue>(key: string, context: EvaluationContext): Promise<ResolutionDetails<T> | undefined> {
+    const cacheKey = this.buildKey(key, context);
+    return Promise.resolve(this.getItem<T>(cacheKey)?.details);
+  }
+
+  del(key: string, context: EvaluationContext): Promise<void> {
+    const cacheKey = this.buildKey(key, context);
+    return Promise.resolve(localStorage.removeItem(cacheKey));
+  }
+
+  flush(flagKey?: string): Promise<void> {
+    const prefix = flagKey ? `${PREXFIX}${flagKey}` : PREXFIX;
+    this.entries().forEach(entry => {
+      if (entry[0].startsWith(prefix)) {
+        localStorage.removeItem(entry[0]);
+      }
+    });
+    return Promise.resolve();
```

Comment:
Is localStorage synchronous? Just checking we don't need to wait for anything here

Reply:
It is' The reason I've made this all async is because I can easily see somebody using an async cache impl, like redis or something... so I've made the interfaces all async consciously.